pkg/cover/cover.go

```diff
@@ -48,3 +60,111 @@ func (cov Cover) Serialize() []uint32 {
 	}
 	return res
 }
+
+const EXT string = ".ko"
+
+func searchModuleName(data []byte) (string, error) {
+	key := []byte("name=")
+	pos := bytes.Index(data, key)
+	if pos == -1 {
+		return "", fmt.Errorf("not find name= keyword")
+	}
+	end := bytes.Index(data[pos:], []byte{0x00})
+	if end == -1 {
+		return "", fmt.Errorf("not find null char")
+	}
+	return string(data[pos+len(key) : pos+end]), nil
+}
+
+func getModuleName(path string) (string, error) {
+	file, err := elf.Open(path)
+	if err != nil {
+		return "", fmt.Errorf("failed to open %v", path)
+	}
+	section := file.Section(".modinfo")
+	if section == nil {
+		return "", fmt.Errorf("no .modinfo section")
+	}
+	data, err := section.Data()
+	if err != nil {
+		return "", fmt.Errorf("failed to read .modinfo")
+	}
+	name, err := searchModuleName(data)
+	if err != nil {
+		section = file.Section(".gnu.linkonce.this_module")
+		if section == nil {
+			return "", fmt.Errorf("no .gnu.linkonce.this_module section")
+		}
+		data, err = section.Data()
+		if err != nil {
+			fmt.Println("failed to read .gnu.linkonce.this_module")
+		}
+		name = string(data)
+	}
+	return name, nil
+}
+
+func walkModulePath(dir string) ([]string, error) {
+	files := []string{}
+	err := filepath.Walk(dir, func(path string, f os.FileInfo, err error) error {
+		if filepath.Ext(path) == EXT {
+			files = append(files, path)
+		}
+		return nil
+	})
+	return files, err
+}
+
+func findModulePaths(objDir string, moduleDirs []string) []string {
+	var files []string
+	var err error
+	if objDir != "" {
+		files, err = walkModulePath(objDir)
+		if err != nil {
+			log.Logf(0, "failed to find modules in %v: %v", objDir, err)
+		}
+	}
+	for _, path := range moduleDirs {
+		mfiles, err := walkModulePath(path)
+		if err != nil {
+			log.Logf(0, "failed to find modules in %v: %v", path, err)
+			continue
+		}
+		files = append(files, mfiles...)
+	}
+	return files
+}
+
+func getKernelModules(objDir string, moduleDirs []string, modules map[string]backend.KernelModule) {
```

Comment:
All of this code somehow needs to be invoked only for Linux. It's Linux-specific.

Reply:
does it mean a elf_linux.go is needed, not sure of it? or just put into backend/elf.go is enough?

Reply:
elf_linux.go won't work. elf_linux.go will be compiled only for linux, but we can have linux target and syz-manager running on windows and elf_linux.go won't be compiled into syz-manager in this case.
I think we can start with putting this code into backend/elf.go and just have an `if` for linux OS and return an errors if we have modules but target OS is not linux.